app/assets/javascripts/discourse/app/components/header.gjs

```diff
@@ -62,6 +66,60 @@ export default class GlimmerHeader extends Component {
     };
   });
 
+  handleFocus = modifierFn((element) => {
+    const panelBody = element.querySelector(PANEL_SELECTOR);
+    if (!panelBody) {
+      return;
+    }
+
+    let isKeyboardEvent = false;
+
+    const handleKeydown = (event) => {
+      if (event.key) {
+        isKeyboardEvent = true;
+      }
+    };
+
+    // avoid triggering focusout on click
+    // otherwise we can double-trigger the menu toggle
+    const handleMousedown = () => {
+      isKeyboardEvent = false;
+    };
+
+    const focusOutHandler = (event) => {
+      if (!isKeyboardEvent) {
+        return;
+      }
+
+      if (!panelBody.contains(event.relatedTarget)) {
+        this.closeCurrentMenu();
+      }
+    };
+
+    panelBody.addEventListener("keydown", handleKeydown);
+    panelBody.addEventListener("mousedown", handleMousedown);
+    panelBody.addEventListener("focusout", focusOutHandler);
+
+    return () => {
+      panelBody.removeEventListener("keydown", handleKeydown);
+      panelBody.removeEventListener("mousedown", handleMousedown);
+      panelBody.removeEventListener("focusout", focusOutHandler);
+    };
+  });
+
+  @action
+  closeCurrentMenu() {
+    if (this.search.visible) {
+      this.toggleSearchMenu();
+    } else if (this.header.userVisible) {
+      this.toggleUserMenu();
+      document.getElementById(USER_BUTTON_ID)?.focus();
+    } else if (this.header.hamburgerVisible) {
+      this.toggleHamburger();
+      document.getElementById(HAMBURGER_BUTTON_ID)?.focus();
+    }
```

Comment:
I don't see any reference to `toggleNavigationMenu`... are we missing an option?